[docs-only] Update the docs helper readme and other related files (#8714)
owncloud/ocis: docs/helpers/README.md

      * [General Extended Envvars Info](#general-extended-envvars-info)
      * [Extract Extended Envvars](#extract-extended-envvars)
      * [Generate Extended Envvar Docs](#generate-extended-envvar-docs)
   * [Tasks for New Releases](#tasks-for-new-releases)
   * [Backporting](#backporting)

## Introduction

> `docs/helpers` contains go programs creating docs by extracting information from the code. The `main.go` program is manually started with `make docs-generate` or via the CI. It calls the other required programs and has these main responsibilities:

Reviewer: it is just one program actually

> `docs/helpers` contains a go program named `main.go` which creates docs by extracting information from the code using additional go programs. Individual steps (programs) can be called manually if needed. Note that not all programs are called automatically by purpose, see the [Tasks for New Releases](#tasks-for-new-releases) below. `main.go` is used by `make docs-generate` which can be called manually or by the CI. It calls the other required programs and has these main responsibilities for automatic runs:

Reviewer 2: ```suggestion
`docs/helpers` contains a go program named `main.go` which creates docs by extracting information from the code using additional go programs. Individual steps (programs) can be called manually if needed. Note that not all programs are called automatically on purpose, see the [Tasks for New Releases](#tasks-for-new-releases) below. `main.go` is used by `make docs-generate` which can be called manually or by the CI. It calls the other required programs and has these main responsibilities for automatic runs:
```

	// DependendServices []Service `yaml:"dependend_services"`
}
```

This yaml file can later be manually edited to add descriptions, default values, etc.

**IMPORTANT**: `RawName`, `Path` and `FoundInCode` are automatically filled by the program. DO NOT EDIT THESE VALUES MANUALLY.

### Generate Extended Envvar Docs

The process further picks up the `yaml` file generated in the `Extract Rogue Envvars` step and renders it to an adoc file (a table is created) using a go template. The template file for this step can be found at `docs/templates/ADOC_extended.tmpl`.

## Tasks for New Releases

> Close before a new release gets published, but **before** a new ocis admin docs branch is created, some tasks need to be done manually. These tasks cant be done automatically! By processing these manual tasks, doc related and important files get updated respectively created, ready to be consumed by the admin docs. The following situations can occur, see the solution provided:

Reviewer 2: ```suggestion
Close before a new release gets published, but **before** a new ocis admin docs branch is created, some tasks need to be done manually. These tasks cant be done automatically! By processing these manual tasks, doc related and important files get updated or created, ready to be consumed by the admin docs. The following situations can occur, see the solution provided:
```

> 1. Admin docs prepares in master for envvar delta changes. Building the admin docs report inexisting referenced delta files.\

Reviewer 2: ```suggestion
1. Admin docs prepares in master for envvar delta changes. Building the admin docs report non-existent referenced delta files.\
```

&#8594; Run the manual tasks described below and restart the branched admin docs.

> 2. A new admin docs branch gets created but no `docs-stable-x.y` branch has been created in the ocis repo so far. Building the branched admin docs report inexisting referenced files.\

Reviewer 2: ```suggestion
2. A new admin docs branch gets created but no `docs-stable-x.y` branch has been created in the ocis repo so far. Building the branched admin docs reports non-existent referenced files.\
```

&#8594; Run the manual tasks described below.\
&#8594; Take care that the `docs-stable-x.y` branch gets created in the ocis repo.\
&#8594; Alternatively you can _temporarily_ point in the branched [docs-ocis/antora.yml](https://github.com/owncloud/docs-ocis/blob/master/antora.yml) `compose_tab_1` to `master`.\
&#8594; Restart building the branched admin docs.

> 3. A new ocis release is close to be released, the manual tasks **have not been** processed so far.\

Reviewer 2: ```suggestion
3. A new ocis release is close to being released, the manual tasks **have not been** processed so far.\
```

&#8594; Run the manual tasks AND backport the results into the `docs-stable-x.y` branch.

> 4. A new ocis release is close to be released, the manual tasks **have been** processed some time ago.\

Reviewer 2: ```suggestion
4. A new ocis release is close to being released, the manual tasks **have been** processed some time ago.\
```

&#8594; Re-run the manual tasks AND backport the results into the `docs-stable-x.y` branch.

5. A new ocis release **has been released**, the manual tasks **have not been** processed so far.\
&#8594; Re-run the manual tasks AND backport the results into the `docs-stable-x.y` branch.
  
### Task List

1. From the ocis root, run `make -C docs docs-generate` and check if there is a change in the `extended-envars.yaml` output.\

> &#8594; In case, process [Extended Envvars](#extended-envvars). When done, re-run the make command and check if the output of `./docs/services/_includes/adoc/extended_configvars.adoc` matches the expectations.

Reviewer 2: ```suggestion
&#8594; In this case, process [Extended Envvars](#extended-envvars). When done, re-run the make command and check if the output of `./docs/services/_includes/adoc/extended_configvars.adoc` matches the expectations.
```

2. In `./docs/helpers` run: `go run . --help` This will give you an overview of available commands. 

>     1. Run `go run . all` to generate respectively update required base files.\

Reviewer 2: ```suggestion
    1. Run `go run . all` to generate and/or update required base files.\
```
